extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQWorkerTaskLauncher.java

```diff
@@ -762,9 +762,11 @@ public boolean didFail()
      */
```

Comment:
Can you please update the Javadoc for this method? Also please mention both the conditions separately and what they guardrail, i.e. `currentTimeMillis - recentFullyStartedWorkerTimeInMs.get() > maxTaskStartDelayMillis` ensures that no new worker has been launched in the last maxTaskStartDelayMillis.

Reply:
Done